sync/src/block/extension.rs

```diff
@@ -276,7 +276,7 @@ impl NetworkExtension for Extension {
                         }
                         _ => {}
                     }
-                    self.dismiss_request(&id, request_id);
+                    request_list.retain(|(i, _)| *i != request_id);
```

Comment:
This one is not a good solution... but let's leave it for now.
We should change this to call function later.

Reply:
Yes, it should be fixed. It's not a good idea...